Modules/Project/Services/ProjectService.php

```diff
@@ -462,13 +462,17 @@ private function formatProjectFTEFOrExportAll($employees)
         return $teamMembers;
     }
 
-    public function getProjectsWithTeamMemberRequirementData()
+    public function getProjectsWithTeamMemberRequirementData($request)
     {
-        $projectsWithTeamMemberRequirement = Project::with('client')->status('active')
+        $query = Project::with('client')->status('active')
             ->withCount('getTeamMembers as team_member_count')
             ->withSum('resourceRequirement as team_member_needed', 'total_requirement')
-            ->havingRaw('team_member_needed - team_member_count')
-            ->get();
+            ->havingRaw('team_member_needed - team_member_count');
+
+        if ($request != null) {
+            $query->where('name', $request);
+        }
+        $projectsWithTeamMemberRequirement = $query->get();
```

Comment:
If you are using the above-suggested code, then please remove this code.

and also one more suggestion in this code. You can write like this if the condition. `empty` is a PHP function that checks whether a variable is empty or not.
```suggestion
        if ((!empty($request->name)) {
            $query->where('name', $request);
        }
```

Reply:
Yes Sir, I'll keep that in mind.